src/login/logind-user.c

```diff
@@ -891,7 +892,13 @@ int config_parse_user_tasks_max(
         assert(rvalue);
         assert(data);
 
-        /* First, try to parse as percentage */
+        /* Unlimited? */
+        if (isempty(rvalue) || streq(rvalue, "infinity")) {
```

Comment:
Yeah, it seems that in general the empty setting is equivalent to not having the setting at all, see e.g. cg_weight_parse, cg_cpu_shares parse. Thus we should do the same, and fall back to the default, i.e. let DefaultTasksMax win.